src/libs/actions/Report.js

```diff
@@ -59,10 +59,6 @@ const allReports = {};
 let conciergeChatReportID;
 const typingWatchTimers = {};
 
-// Map of optimistic report action IDs. These should be cleared when replaced by a recent fetch of report history
-// since we will then be up to date and any optimistic actions that are still waiting to be replaced can be removed.
-const optimisticReportActionIDs = {};
```

Comment:
We were:

- adding all optimistic actions `clientID` to an array on each report object
- removing them later whichever one happens first
    - The real action created in the server shows up in Pusher
    - The latest actions for the report are fetched

Clearing out the actions was done to take care of an edge case where:

- A report comment would fail to be created in the server
- No error message would be displayed and there would be no way to dismiss the message
- These comments would get "stuck" in a kind of limbo of never being created and unable to be removed 
- So to counteract that we would clear these out when fetching new actions

Ultimately, I think this is not as much of a problem any more since we are handling the failure cases and giving the user a way to dismiss a message that could not be created for some reason.

Reply:
Makes sense. Thanks for the explanation.